pallets/clusters/src/lib.rs

```diff
@@ -269,23 +279,21 @@ pub mod pallet {
 			let role_hash = blake2_128(&[&cluster_id.as_slice(), role_name.as_slice()].concat());
 
 			// Check that the caller is the owner of the cluster
-			let cluster = <Clusters<T>>::get(&cluster_id).ok_or(Error::<T>::ClusterNotFound)?;
-			ensure!(who == cluster.owner, Error::<T>::NoPermission);
+			ensure!(who == <Clusters<T>>::get(&cluster_id).ok_or(Error::<T>::ClusterNotFound)?.owner, Error::<T>::NoPermission);
```

Comment:
why this refactor? you are using `cluster` also under, so getting out once would have helped

Reply:
Right..